src/entities/trades/swapr-v3/SwaprV3.ts

```diff
@@ -101,26 +107,18 @@ export class SwaprV3Trade extends TradeWithSwapTransaction {
           quoteCurrency.name,
         )
 
-    invariant(
-      (await provider.getNetwork()).chainId == chainId,
-      `SwaprV3Trade.getQuote: currencies chainId does not match provider's chainId`,
-    )
-
-    const routes = await getRoutes(tokenIn, tokenOut, chainId)
+    const routes = await getRoutes(setToken, quoteToken, chainId)
 
     const fee =
       routes?.length > 0 && routes[0].pools.length > 0
         ? new Percent(routes[0].pools[0].fee.toString(), '1000000')
         : new Percent('0', '0')
```

Comment:
Let's use 1 instead of 0. Otherwise this will break.

```suggestion
        : new Percent('0', '1')
```